iroh-net/src/magic_endpoint.rs

```diff
@@ -287,6 +289,15 @@ impl MagicEndpoint {
         self.msock.my_derp().await
     }
 
+    /// Get information on all the peers we have connection information about.
+    ///
+    /// Includes the node's [`PublicKey`], potential DERP region, it's addresses with any known
+    /// latency, and its [`crate::magicsock::ConnectionType`], which let's us know if we are currently communicating
+    /// with that node over a `Direct` (UDP) or `Relay` (DERP) connection.
+    pub async fn node_infos(&self) -> anyhow::Result<Vec<NodeInfo>> {
```

Comment:
nit: maybe `active_connections` or so? `node_infos` to me sounds a bit more static than what this method does. but also not sure myself, it's fine to leave it as is.

Reply:
I did some reading. We *do* delete endpoints, but it is currently only triggered by user action. Any time we `set_network_map`, we remove any connections that were made to us "in the wild". 

But the only method that calls `set_network_map` is `add_known_addrs`. Which... is an interesting side effect that we haven't really surfaced. I'm not sure that we intended to have every time we add a new address to the netmap, that it should clear all connections made by other nodes to us.

However, since this happens only on user action, we are likely carrying around a bunch of "inactive" connections, so until / unless we change this, I don't think `active_connections` is the best name.

@b5 I think `ConnectionInfo` might be the right name for the `EndpointInfo` outside of the `MagicSock`.

which would make this `connection_infos` & I will add the above context into the function docs for both `connection_infos` and `add_known_addrs`.